File: hotel-booking/src/main/java/nostra/cosa/hotelbooking/data/entity/Accommodation.java
package nostra.cosa.hotelbooking.data.entity;

import java.util.List;
import java.util.Set;

import jakarta.persistence.*;
import lombok.*;
import nostra.cosa.hotelbooking.data.entity.enums.AccommodationType;
import nostra.cosa.hotelbooking.data.entity.enums.ServiceType;

/**
 * Accommodation data layer representation.
 */
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Data
@Table(name = "accommodation_table")
public class Accommodation{

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(name = "authentication_id", nullable = false)
  private Long authenticationId;

  @Column(name = "accommodation_name", nullable = false)
  private String accommodationName;

  @Column(name = "address_id", nullable = false)
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long addressId;

  @Column(name = "email_address", nullable = false)
  private String emailAddress;

  @Column(name = "phone_number", nullable = false)
  private String phoneNumber;

  @Column(name = "accommodation_type", nullable = false)
  private AccommodationType accommodationType;

  @Column(name = "service_types", nullable = false)
  private List<ServiceType> serviceTypes;

  @Column(name = "room_ids", nullable = false)
  private Set<Integer> roomIds;

}

File: client/src/types/Accommodation.ts
import { Address } from "./Address";
import { AuthenticationData } from "./AuthenticationData";
import { Room } from "./Room";
import { AccommodationType } from "./enums/AccommodationType";
import { ServiceType } from "./enums/ServiceType";

export type Accommodation = {
  id: number;
  authenticationData: AuthenticationData;
  accommodationName: string;
  address: Address;
  emailAddress: string;
  phoneNumber: string;
  accommodationType: AccommodationType;
  serviceTypes: ServiceType;
  rooms: Room[];
}